core/vm/privacy/bulletproof.go

```diff
@@ -233,7 +233,14 @@ func HashPointsToBytes(points []ECPoint) []byte {
 		input = append(input, pointInByte...)
 	}
 
-	return crypto.Keccak256(input)
+	hashedInt, err := HashToScalar(input, curve)
+	if err != nil {
+		check(err)
```

Comment:
Why call function `check` here? then panic?

Reply:
yes, because all the functions that calls this has no error handling
then I saw check() being used in similar way in this file